### ISSUE TITLE
ConstNode GetAsXXXX methods refactor

### ISSUE THREAD

**Issue by jingchen2222** [4paradigm/OpenMLDB#580]
**Is your feature request related to a problem? Please describe.**

This issue is going to to Refactor ConstNode's GetAsXXXX methods, e.g. `GetAsString()`, `GetAsDouble` etc, since it is hard to know if these GetAsXXXX methods go well or not. So we have to re-design the methods as follows:
```
bool GetAsInt(int* output);
bool GetAsString(std::string* str)
```

Also, we will support GetAsString for types like `kBool`, `kDate`, `kTimestamp`.
1. kBool: `true` -> `"true"`, kBool: `false` -> `"false"`
2. kTimestamp: `1590115420000` -> `"1590115420000L"`
3. kDate:
```
Date date(2020, 05, 22);
to_string(date.date_)
```
need to discuss what `kDate` string looks like. `yyyy-mm-dd` pattern string will be far more reasonable. 

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by aceforeverd** [comment]
- provide a `Formater` as optional parameter to GetAsString, or just leave as `yyyy-mm-dd` ?
- is it also support e.g `GetAsDouble` for `int` type ?

**Comment by tobegit3hub** [comment]
Refer to https://github.com/4paradigm/OpenMLDB/issues/455#issuecomment-942909834 , we may need to handle null value and different format for data type. More optional parameter may be needed like `Formater` above.

**Comment by jingchen2222** [comment]
> * provide a `Formater` as optional parameter to GetAsString, or just leave as `yyyy-mm-dd` ?
> * is it also support e.g `GetAsDouble` for `int` type ?

Yes, and we have `double GetDouble()` method to get value strictly via its datatype.

I think we can remove methods like GetAsInt, GetAsDouble, GetAsFloat, GetAsInt64, etc, from ConstNode.
Firstly, these methods are badly designed. At least, we need to know GetAsXXXX run success or not. 
Secondly, it's unsafe to GetAsInt64 for double type.  Maybe we can just remove them since the user might use them in the wrong way.

**Comment by jingchen2222** [comment]
> Refer to [#455 (comment)](https://github.com/4paradigm/OpenMLDB/issues/455#issuecomment-942909834) , we may need to handle null value and different format for data type. More optional parameter may be needed like `Formater` above.

We will discuss it. It is not necessary to be methods of ConstNode. 
For example, `SELECT INTO ...` format date value instead of ConstNode.
